pandas/core/resample.py

```diff
@@ -405,6 +405,24 @@ def _groupby_and_aggregate(self, how, grouper=None, *args, **kwargs):
         result = self._apply_loffset(result)
         return self._wrap_result(result)
 
+    def _try_aggregate(self, grouped, how, *args, **kwargs):
+        """
+        Tries to aggregate a 'grouped' object.
+        When 'how' param is a callable, we aggregate item by item
+        to check if the function is reducing or not.
+
+        Parameters
+        ----------
+        grouped : GroupBy object
+        how : string / cython mapped function
+        """
+        if not compat.callable(how) or isinstance(grouped, PanelGroupBy):
```

Comment:
instead of this, add the PanelGroupbyCase above (IOW it can only be a panel groupby in the except clause of the previous try/except).

you can then put the rest in the try/except, no need to add another function here. add comments indicating what is going on.